netket/machine/jax.py

```diff
@@ -332,3 +332,95 @@ def JaxRbm(hilbert, alpha, dtype=complex):
         stax.serial(stax.Dense(alpha * hilbert.size), LogCoshLayer, SumLayer()),
         dtype=dtype,
     )
+
+
+def JaxMpsPeriodic(hilbert, bond_dim, diag=False, symperiod=-1, dtype=complex):
+    return Jax(hilbert, stax.serial(MpsPeriodicLayer(hilbert, bond_dim, diag, symperiod, dtype)), dtype=dtype)
+
+
+def MpsPeriodicLayer(hilbert, bond_dim, diag=False, symperiod=-1, dtype=complex):
+    #default standard deviation equals 1e-2
+    normal_init = jax.nn.initializers.normal()
+
+    L = hilbert.size
+    phys_dim = hilbert.local_size
+    diag = diag
+    dtype = dtype
+
+    #determine transformation from local states to indices
+    local_states = jax.numpy.array(hilbert.local_states)
+    loc_vals_spacing = jax.numpy.roll(local_states, -1)[0:-1] - local_states[0:-1]
+    if jax.numpy.max(loc_vals_spacing) == jax.numpy.min(loc_vals_spacing):
+        loc_vals_spacing = loc_vals_spacing[0]
+    else:
+        raise AssertionError("JaxMpsPeriodic can only be used with evenly spaced hilbert local values")
```

Comment:
I think this is not a great limitation, as you say most of the interesting cases are evenly spaced...!